tests/components/energyid/test_config_flow.py

```diff
@@ -0,0 +1,775 @@
+            result["flow_id"],
+            {
+                CONF_PROVISIONING_KEY: TEST_PROVISIONING_KEY,
+                CONF_PROVISIONING_SECRET: TEST_PROVISIONING_SECRET,
+            },
+        )
+        await hass.async_block_till_done()
+
+        assert result2["type"] is FlowResultType.CREATE_ENTRY
+        assert result2["title"] == TEST_RECORD_NAME
+        assert result2["data"][CONF_PROVISIONING_KEY] == TEST_PROVISIONING_KEY
+        assert result2["data"][CONF_PROVISIONING_SECRET] == TEST_PROVISIONING_SECRET
+        assert result2["description"] == "add_sensor_mapping_hint"
+
+
+@pytest.mark.parametrize("claimed", [False])
+async def test_config_flow_user_step_needs_claim(
+    hass: HomeAssistant, claimed: bool
+) -> None:
+    """Test user step where device needs to be claimed."""
+    mock_client = MagicMock()
+    mock_client.authenticate = AsyncMock(return_value=claimed)
+    mock_client.get_claim_info.return_value = {"claim_url": "http://claim.me"}
+
+    with (
+        patch(
+            "homeassistant.components.energyid.config_flow.WebhookClient",
+            return_value=mock_client,
+        ),
+        patch("homeassistant.components.energyid.config_flow.asyncio.sleep"),
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_PROVISIONING_KEY: TEST_PROVISIONING_KEY,
+                CONF_PROVISIONING_SECRET: TEST_PROVISIONING_SECRET,
+            },
+        )
+        assert result2["type"] is FlowResultType.EXTERNAL_STEP
+        assert result2["step_id"] == "auth_and_claim"
+
+
+async def test_config_flow_auth_and_claim_step_success(hass: HomeAssistant) -> None:
+    """Test auth_and_claim step where the device becomes claimed after polling."""
+    mock_unclaimed_client = MagicMock()
+    mock_unclaimed_client.authenticate = AsyncMock(return_value=False)
+    mock_unclaimed_client.get_claim_info.return_value = {"claim_url": "http://claim.me"}
+
+    mock_claimed_client = MagicMock()
+    mock_claimed_client.authenticate = AsyncMock(return_value=True)
+    mock_claimed_client.recordNumber = TEST_RECORD_NUMBER
+    mock_claimed_client.recordName = TEST_RECORD_NAME
+
+    call_count = 0
+
+    def mock_webhook_client(*args, **kwargs):
+        nonlocal call_count
+        call_count += 1
+        if call_count == 1:
+            return mock_unclaimed_client
+        return mock_claimed_client
+
+    with (
+        patch(
+            "homeassistant.components.energyid.config_flow.WebhookClient",
+            side_effect=mock_webhook_client,
+        ),
+        patch("homeassistant.components.energyid.config_flow.asyncio.sleep"),
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
```

Comment:
Check the result is as expected, don't just assign and discard. Same for the other tests